### ISSUE TITLE
Error in `GET()` when deploy the app on shinyapps.io 

### ISSUE THREAD

**Issue by michal-kinel** [rstudio/rsconnect#1020]
Hi, I'm not 100% sure if the error is related to rsconnect or to the shinyapps server itself. When connecting with rsconnect::setAccountInfo with any problem, I do the deply of the app (locally has no problems). 

The console returns the following error message:

```
  building: Installing system dependencies
## Begin Task Log ###############################################
Error in `GET()`:
! <https://api.shinyapps.io/v1/tasks/******/logs/>
  failed with HTTP status 404
Not Found
```
The backtrace stored in the last error:

```
> rlang::last_trace(drop = FALSE)
<error/rsconnect_http_404>
Error in `GET()`:
! <https://api.shinyapps.io/v1/tasks/*****/logs/>
  failed with HTTP status 404
Not Found
---
Backtrace:
     ▆
  1. └─rsconnect::deployApp(...)
  2.   └─client$waitForTask(taskId, quiet)
  3.     └─rsconnect::taskLog(taskId, authInfo$name, authInfo$server, output = "stderr")
  4.       ├─base::cat(client$getTaskLogs(taskId), file = conn)
  5.       └─client$getTaskLogs(taskId)
  6.         └─rsconnect:::GET(service, authInfo, path)
  7.           └─rsconnect:::httpRequest(...)
  8.             └─rsconnect:::handleResponse(httpResponse, error_call = error_call)
  9.               └─rsconnect (local) reportError(unlist(json$error))
 10.                 └─cli::cli_abort(...)
 11.                   └─rlang::abort(...)
```

Session info:

```
R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/atlas/libblas.so.3.10.3 
LAPACK: /usr/lib/x86_64-linux-gnu/atlas/liblapack.so.3.10.3;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=es_ES.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=C                  LC_COLLATE=es_ES.UTF-8    
 [5] LC_MONETARY=es_ES.UTF-8    LC_MESSAGES=es_ES.UTF-8   
 [7] LC_PAPER=es_ES.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=es_ES.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/Madrid
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

other attached packages:
[1] rsconnect_1.1.1.9000

loaded via a namespace (and not attached):
 [1] digest_0.6.33       assertthat_0.2.1    R6_2.5.1           
 [4] lubridate_1.9.3     RApiSerialize_0.1.2 RcppParallel_5.1.7 
 [7] timechange_0.2.0    generics_0.1.3      lifecycle_1.0.3    
[10] packrat_0.9.2       cli_3.6.1           askpass_1.2.0      
[13] openssl_2.1.1       compiler_4.3.1      stringfish_0.15.8  
[16] rstudioapi_0.15.0   tools_4.3.1         curl_5.1.0         
[19] Rcpp_1.0.11         BiocManager_1.30.22 crayon_1.5.2       
[22] jsonlite_1.8.7      rlang_1.1.1         qs_0.25.5          
[25] keyring_1.3.1      
```

**Comment by aronatkins** [comment]
Thanks so much for reporting this issue. We've discovered that a change to shinyapps.io started returning a 404 for some task-log requests when previously it would return an empty result.

@mslynch is going to work on resolving this problem and will report back once the fix is in place.

**Comment by michal-kinel** [comment]
Hi, sorry @mslynch, I don't know if you're going to know what's wrong with this error. If it's my machine or I need to update something.


 ```
 building: Installing system dependencies
## Begin Task Log ####################
Quarto not found.
[2023-10-17T18:52:38.093058881+0000] Execute script: packages/build/sodium.sh
+ set -e
+ apt-get update -qq
W: http://cran.rstudio.com/bin/linux/ubuntu/jammy-cran40/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
+ apt-get install -y libsodium-dev
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  libsodium23
The following NEW packages will be installed:
  libsodium-dev libsodium23
0 upgraded, 2 newly installed, 0 to remove and 24 not upgraded.
Need to get 349 kB of archives.
After this operation, 1229 kB of additional disk space will be used.
Get:1 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy/main amd64 libsodium23 amd64 1.0.18-1build2 [164 kB]
Get:2 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy/main amd64 libsodium-dev amd64 1.0.18-1build2 [185 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 349 kB in 0s (13.0 MB/s)
Selecting previously unselected package libsodium23:amd64.	
(Reading database ... 	(Reading database ... 5%	(Reading database ... 10%	(Reading database ... 15%	(Reading database ... 20%	(Reading database ... 25%	(Reading database ... 30%	(Reading database ... 35%	(Reading database ... 40%	(Reading database ... 45%	(Reading database ... 50%	(Reading database ... 55%	(Reading database ... 60%	(Reading database ... 65%	(Reading database ... 70%	(Reading database ... 75%	(Reading database ... 80%	(Reading database ... 85%	(Reading database ... 90%	(Reading database ... 95%	(Reading database ... 100%	(Reading database ... 289117 files and directories currently installed.)	
Preparing to unpack .../libsodium23_1.0.18-1build2_amd64.deb ...	
Unpacking libsodium23:amd64 (1.0.18-1build2) ...	
Selecting previously unselected package libsodium-dev:amd64.	
Preparing to unpack .../libsodium-dev_1.0.18-1build2_amd64.deb ...	
Unpacking libsodium-dev:amd64 (1.0.18-1build2) ...	
Setting up libsodium23:amd64 (1.0.18-1build2) ...	
Setting up libsodium-dev:amd64 (1.0.18-1build2) ...	
Processing triggers for libc-bin (2.35-0ubuntu3.1) ...	
[2023-10-17T18:52:41.227229869+0000] Execute script: packages/build/units.sh
+ set -e
+ apt-get update -qq
W: http://cran.rstudio.com/bin/linux/ubuntu/jammy-cran40/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
+ apt-get install -y libudunits2-dev
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  libudunits2-0 libudunits2-data
The following NEW packages will be installed:
  libudunits2-0 libudunits2-data libudunits2-dev
0 upgraded, 3 newly installed, 0 to remove and 24 not upgraded.
Need to get 433 kB of archives.
After this operation, 1064 kB of additional disk space will be used.
Get:1 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy/universe amd64 libudunits2-data all 2.2.28-3 [19.0 kB]
Get:2 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy/universe amd64 libudunits2-0 amd64 2.2.28-3 [50.3 kB]
Get:3 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy/universe amd64 libudunits2-dev amd64 2.2.28-3 [364 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 433 kB in 0s (13.7 MB/s)
Selecting previously unselected package libudunits2-data.	
(Reading database ... 	(Reading database ... 5%	(Reading database ... 10%	(Reading database ... 15%	(Reading database ... 20%	(Reading database ... 25%	(Reading database ... 30%	(Reading database ... 35%	(Reading database ... 40%	(Reading database ... 45%	(Reading database ... 50%	(Reading database ... 55%	(Reading database ... 60%	(Reading database ... 65%	(Reading database ... 70%	(Reading database ... 75%	(Reading database ... 80%	(Reading database ... 85%	(Reading database ... 90%	(Reading database ... 95%	(Reading database ... 100%	(Reading database ... 289195 files and directories currently installed.)	
Preparing to unpack .../libudunits2-data_2.2.28-3_all.deb ...	
Unpacking libudunits2-data (2.2.28-3) ...	
Selecting previously unselected package libudunits2-0.	
Preparing to unpack .../libudunits2-0_2.2.28-3_amd64.deb ...	
Unpacking libudunits2-0 (2.2.28-3) ...	
Selecting previously unselected package libudunits2-dev.	
Preparing to unpack .../libudunits2-dev_2.2.28-3_amd64.deb ...	
Unpacking libudunits2-dev (2.2.28-3) ...	
Setting up libudunits2-data (2.2.28-3) ...	
Setting up libudunits2-0 (2.2.28-3) ...	
Setting up libudunits2-dev (2.2.28-3) ...	
Processing triggers for install-info (6.8-4build1) ...	
Processing triggers for libc-bin (2.35-0ubuntu3.1) ...	
[2023-10-17T18:52:44.236825335+0000] Execute script: packages/build/keyring.sh
+ set -e
+ apt-get update -qq
W: http://cran.rstudio.com/bin/linux/ubuntu/jammy-cran40/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
+ apt-get install -y libsecret-1-dev libsodium-dev
Reading package lists...
Building dependency tree...
Reading state information...
libsodium-dev is already the newest version (1.0.18-1build2).
The following additional packages will be installed:
  gir1.2-secret-1 libgcrypt20-dev libgpg-error-dev libsecret-1-0
  libsecret-common
Suggested packages:
  libgcrypt20-doc
The following NEW packages will be installed:
  gir1.2-secret-1 libgcrypt20-dev libgpg-error-dev libsecret-1-0
  libsecret-1-dev libsecret-common
0 upgraded, 6 newly installed, 0 to remove and 24 not upgraded.
Need to get 973 kB of archives.
After this operation, 7093 kB of additional disk space will be used.
Get:1 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy/main amd64 libsecret-common all 0.20.5-2 [4278 B]
Get:2 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy/main amd64 libsecret-1-0 amd64 0.20.5-2 [124 kB]
Get:3 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy/main amd64 gir1.2-secret-1 amd64 0.20.5-2 [9318 B]
Get:4 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy/main amd64 libgpg-error-dev amd64 1.43-3 [129 kB]
Get:5 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy/main amd64 libgcrypt20-dev amd64 1.9.4-3ubuntu3 [572 kB]
Get:6 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy/main amd64 libsecret-1-dev amd64 0.20.5-2 [135 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 973 kB in 0s (27.9 MB/s)
Selecting previously unselected package libsecret-common.	
(Reading database ... 	(Reading database ... 5%	(Reading database ... 10%	(Reading database ... 15%	(Reading database ... 20%	(Reading database ... 25%	(Reading database ... 30%	(Reading database ... 35%	(Reading database ... 40%	(Reading database ... 45%	(Reading database ... 50%	(Reading database ... 55%	(Reading database ... 60%	(Reading database ... 65%	(Reading database ... 70%	(Reading database ... 75%	(Reading database ... 80%	(Reading database ... 85%	(Reading database ... 90%	(Reading database ... 95%	(Reading database ... 100%	(Reading database ... 289235 files and directories currently installed.)	
Preparing to unpack .../0-libsecret-common_0.20.5-2_all.deb ...	
Unpacking libsecret-common (0.20.5-2) ...	
Selecting previously unselected package libsecret-1-0:amd64.	
Preparing to unpack .../1-libsecret-1-0_0.20.5-2_amd64.deb ...	
Unpacking libsecret-1-0:amd64 (0.20.5-2) ...	
Selecting previously unselected package gir1.2-secret-1:amd64.	
Preparing to unpack .../2-gir1.2-secret-1_0.20.5-2_amd64.deb ...	
Unpacking gir1.2-secret-1:amd64 (0.20.5-2) ...	
Selecting previously unselected package libgpg-error-dev.	
Preparing to unpack .../3-libgpg-error-dev_1.43-3_amd64.deb ...	
Unpacking libgpg-error-dev (1.43-3) ...	
Selecting previously unselected package libgcrypt20-dev.	
Preparing to unpack .../4-libgcrypt20-dev_1.9.4-3ubuntu3_amd64.deb ...	
Unpacking libgcrypt20-dev (1.9.4-3ubuntu3) ...	
Selecting previously unselected package libsecret-1-dev:amd64.	
Preparing to unpack .../5-libsecret-1-dev_0.20.5-2_amd64.deb ...	
Unpacking libsecret-1-dev:amd64 (0.20.5-2) ...	
Setting up libgpg-error-dev (1.43-3) ...	
Setting up libsecret-common (0.20.5-2) ...	
Setting up libsecret-1-0:amd64 (0.20.5-2) ...	
Setting up gir1.2-secret-1:amd64 (0.20.5-2) ...	
Setting up libgcrypt20-dev (1.9.4-3ubuntu3) ...	
Setting up libsecret-1-dev:amd64 (0.20.5-2) ...	
Processing triggers for libc-bin (2.35-0ubuntu3.1) ...	
Processing triggers for install-info (6.8-4build1) ...	
[2023-10-17T18:52:47.442006147+0000] Execute script: packages/build/sf.sh
+ set -e
++ lsb_release -cs
+ OS_CODENAME=jammy
+ '[' jammy == xenial ']'
+ apt-get update -qq
W: http://cran.rstudio.com/bin/linux/ubuntu/jammy-cran40/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
+ apt-get install -y libudunits2-dev libgdal-dev libgeos-dev libproj-dev
Reading package lists...
Building dependency tree...
Reading state information...
libgdal-dev is already the newest version (3.4.1+dfsg-1build4).
libgeos-dev is already the newest version (3.10.2-1).
libgeos-dev set to manually installed.
libproj-dev is already the newest version (8.2.1-1).
libudunits2-dev is already the newest version (2.2.28-3).
0 upgraded, 0 newly installed, 0 to remove and 24 not upgraded.
[2023-10-17T18:52:49.673355771+0000] Execute script: packages/build/RSelenium.sh
+ set -e
+ apt-get update -qq
W: http://cran.rstudio.com/bin/linux/ubuntu/jammy-cran40/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
+ DEBIAN_FRONTEND=noninteractive
+ apt-get install -y libssl-dev phantomjs
Reading package lists...
Building dependency tree...
Reading state information...
Package phantomjs is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
E: Package 'phantomjs' has no installation candidate
## End Task Log #################
```

**Comment by mslynch** [comment]
@michal-kinel That looks like a bug. I've filed another issue here to track: https://github.com/rstudio/shinyapps-package-dependencies/issues/359
